src/providers/ldap/sdap_async_users.c

```diff
@@ -278,8 +278,10 @@ int sdap_save_user(TALLOC_CTX *memctx,
                                    &samaccountname);
         if (ret == EOK) {
             ret = ENOENT;
-            new_upn = talloc_asprintf(memctx, "%s@%s", samaccountname,
-                                      dom->realm);
+            if (dom->realm != NULL) {
```

Comment:
In what case can it happen than `dom->realm == NULL`?

Reply:
Hi,

this happens e.g. if AD is accessed with `id_provider = ldap`, see test.

bye,
Sumit

Reply:
Commit message says:
```
If 'id_provider = ldap' is used to access Active Directory the realm might
not be set in the internal domain data
```

Does "might" mean that it also may be set even if 'id_provider = ldap'?
If so, is it ok to synthesize canonical principal in this case?

Reply:
Hi,

yes, since this only happens if the LDAP schema is `SDAP_SCHEMA_AD` we can imo assume an AD environment where this canonical principal makes sense.

bye,
Sumit